ArtOfIllusion/src/artofillusion/material/LinearMaterialMapping.java

```diff
@@ -45,7 +45,8 @@ public String getName()
     return "Linear";
   }
 
-  public static boolean legalMapping(Object3D obj, Material mat)
+  @Override
+  public boolean legalMapping(Object3D obj, Material mat)
```

Comment:
I sort of get where you are going with this, but these are an API-breaking change. Any plugins or scripts that call `XYZMapping.legalMapping()` would be broken if this change was released. We're not really in a position to fix that breakage quire yet, so I want to approach this more carefully.

We also need to make a larger design decision here: Are we going to keep our list of mappings as `Class` objects, or as instances of the specific mapping types? (See also #24, and the differences in how parts of that list are generated) In general, I think that moving away from reflection, if there are clean ways of doing without, would be preferable.